src/Raven.Server/Documents/Handlers/Debugging/DebugPackage/Analyzers/NetworkInfoAnalyzer.cs

```diff
@@ -0,0 +1,173 @@
+                    new DetectedIssue("High number of established TCP connections",
+                        $"There are {established.NumberOfConnectionsInState} established TCP connections in total",
+                        IssueSeverity.Warning, IssueCategory.Server)
+                    {
+                        RecommendedAction = "Please review top remote addresses that are establishing TCP connections to RavenDB server"
+                    });
+            }
+
+            if (established.TopConnectionsInState.Count > 0)
+            {
+                var topConnection = established.TopConnectionsInState.OrderByDescending(x => x.Value).FirstOrDefault();
+
+                if (topConnection.Value > 300)
+                {
+                    issues.ServerIssues.Add(
+                        new DetectedIssue("High number of established TCP connections",
+                            $"There are {topConnection.Value} established TCP connections coming from IP {topConnection.Key}",
+                            IssueSeverity.Error, IssueCategory.Server)
+                        {
+                            RecommendedAction = "This might be an indication that you are creating a large number of DocumentStore instances. " +
+                                                "Are you creating a DocumentStore per request, instead of using DocumentStore as a singleton? ",
+                        });
+                }
+            }
+        }
+
+        if (timeWait != null)
+        {
+            if (timeWait.NumberOfConnectionsInState > 300)
+            {
+                issues.ServerIssues.Add(
+                    new DetectedIssue("High number of TCP connections in TIME_WAIT state",
+                        $"There are {timeWait.NumberOfConnectionsInState} TCP connections in TIME_WAIT state",
+                        IssueSeverity.Warning, IssueCategory.Server));
+            }
+
+            if (established != null && timeWait.NumberOfConnectionsInState > established.NumberOfConnectionsInState / 2)
+            {
+                issues.ServerIssues.Add(
+                    new DetectedIssue("High number of TCP connections in TIME_WAIT state",
+                        $"There are {timeWait.NumberOfConnectionsInState} TCP connections in TIME_WAIT state while " +
+                        $"the number of ESTABLISHED connections is {established.NumberOfConnectionsInState}. " +
+                        "This could indicate a high volume of short-lived connections while it is generally expected " +
+                        "to have a stable number of established connections with proper connection reuse.",
+                        IssueSeverity.Warning, IssueCategory.Server));
+            }
+        }
+
+        if (NetworkInfo.PingTestResults != null)
+        {
+            foreach (var pingResult in NetworkInfo.PingTestResults)
+            {
+                if (pingResult.TcpInfo.ReceiveTime > 2000)
```

Comment:
Shouldn't you also include the election timeout here?

Reply:
The comparison of ping times and comparison to election timeout is done in `DetectClusterWideIssues()`. Here we analyze each node independently.